### PR TITLE
[EDCO-15] Header Component

### PR DESCRIPTION
**Pull request by mmoio**
## Description

Adds [Header Component](https://www.figma.com/file/6Hn7Wxmvg6eCSqsMrIP7VU/CMS---UI-Components-%26-Templates?node-id=1%3A16135&t=nzKrnFmQ3X5gOCVE-1) & related stories

Fixes # (edco-15)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] Storybook
- [x] Visual testing

### CODE REVIEW

**Comment by mmoio** [comment]
Note that the dropdown and mobile designs are currently missing. The dark theme is also missing but I added it by replicating the design of the other components.

**Comment by stratoivandiluccio** [comment]
> Note that the dropdown and mobile designs are currently missing. The dark theme is also missing but I added it by replicating the design of the other components.

I see some update on Figma about the Header Component. I think now you have more elements to work on.

https://www.figma.com/file/6Hn7Wxmvg6eCSqsMrIP7VU/CMS---UI-Components-%26-Templates?node-id=1084-71992&t=qWePooSLePxT4gGx-0

**Comment by mmoio** [comment]
Added a default mobile and desktop Dropdown item in addition to accepting any jsx element and refactored the DialogBubble as a separate component

**Review by vito80ba — CHANGES_REQUESTED** [review]
Some considerations and tuning:
- the name of website must be in bold (font-weight 700)
- increase vertical padding of the main _div_
- increase vertical padding within the items of submenu

File: src/components/Ctas.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import Stack from '@mui/material/Stack';
+import { isValidElement } from 'react';
+import { type CommonProps } from 'types/components';
+
+interface CtaButtonProps extends Partial<ButtonProps> {
+  text: string;
+}
+
+type CtaButton = CtaButtonProps | JSX.Element;
+
```
Comment (stratoivandiluccio) [inline src/components/Ctas.tsx:11]: Please, let's use the `Generic` type exported from 'src/types/components' instead `JSX.Element`

File: src/components/Ctas.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export interface CtaProps extends CommonProps {
+  ctaButtons?: CtaButton[];
+  reverse?: boolean;
+}
+
+const isButtonProps = (button: CtaButton): button is CtaButtonProps => {
```
Comment (stratoivandiluccio) [inline src/components/Ctas.tsx:17]: I defined a generic function utility with the same purpose in 'src/utils'.

**Review by stratoivandiluccio — APPROVED** [review]

**Pull request merged**